pkg/api/handlers/poweruserworkspace.go

```diff
@@ -148,6 +148,104 @@ func (p *PowerUserWorkspaceHandler) ListAllServers(req api.Context) error {
 	})
 }
 
+func (p *PowerUserWorkspaceHandler) ListAllServersForAllEntries(req api.Context) error {
+	// Get all workspaces
+	var workspaceList v1.PowerUserWorkspaceList
+	if err := req.List(&workspaceList); err != nil {
+		return fmt.Errorf("failed to list power user workspaces: %w", err)
+	}
+
+	// Get all entries from all workspaces
+	var allEntries []v1.MCPServerCatalogEntry
+	var entryList v1.MCPServerCatalogEntryList
+	if err := req.List(&entryList); err != nil {
+		return fmt.Errorf("failed to list entries: %w", err)
+	}
+
+	// Create a map of workspace names for efficient lookup
+	workspaceNames := make(map[string]bool)
+	for _, workspace := range workspaceList.Items {
+		workspaceNames[workspace.Name] = true
+	}
```

Comment:
This can be initialized with a size and it's idiomatic to use the empty struct instead of bool for sets in go:



```suggestion
	workspaceNames := make(map[string]struct{}, len(workspaceList.Items))
	for _, workspace := range workspaceList.Items {
		workspaceNames[workspace.Name] = struct{}{}
	}
```